pkg/pgclient/client.go

```diff
@@ -143,40 +167,49 @@ func NewClientWithPool(cfg *Config, numCopiers int, connPool *pgxpool.Pool, mt t
 		InvertedLabelsCacheSize: cfg.CacheConfig.InvertedLabelsCacheSize,
 	}
 
-	labelsReader := lreader.NewLabelsReader(dbConn, labelsCache, mt.ReadAuthorizer())
+	readerConnPool := pgxconn.NewPgxConn(readerPool)
+	connPools := []pgxconn.PgxConn{readerConnPool}
+	labelsReader := lreader.NewLabelsReader(readerConnPool, labelsCache, mt.ReadAuthorizer())
 	exemplarKeyPosCache := cache.NewExemplarLabelsPosCache(cfg.CacheConfig)
 
-	dbQuerierConn := pgxconn.NewQueryLoggingPgxConn(connPool)
+	dbQuerierConn := pgxconn.NewQueryLoggingPgxConn(readerPool)
 	dbQuerier := querier.NewQuerier(dbQuerierConn, metricsCache, labelsReader, exemplarKeyPosCache, mt.ReadAuthorizer())
 	queryable := query.NewQueryable(dbQuerier, labelsReader)
 
 	var dbIngestor *ingestor.DBIngestor
 	if !readOnly {
 		var err error
-		dbIngestor, err = ingestor.NewPgxIngestor(dbConn, metricsCache, seriesCache, exemplarKeyPosCache, &c)
+		writerConnPool := pgxconn.NewPgxConn(writerPool)
+		dbIngestor, err = ingestor.NewPgxIngestor(writerConnPool, metricsCache, seriesCache, exemplarKeyPosCache, &c)
```

Comment:
Isn't this a little bit dangerous in that it might panic if someone calls `client.Ingest()` on a readonly client?

Reply:
well, adding a check of readOnly seems anti-pattern. Instead, we should make 2 interfaces: one for read and one for write, and then implement as said [here](https://github.com/timescale/promscale/pull/1451#discussion_r911999677). That will be a good design.